src/main.py

```diff
@@ -39,128 +56,170 @@ async def on_close() -> None:
         await bc.finish(say_goodbye=False)
 
 
-@client.event
-async def on_message(message: Message) -> None:
-    global controllers
-    global loaded_image
+# Allow only one async routine to calculate list at a time
+list_task_control_lock = asyncio.Lock()
+
 
-    if message.author == client.user:
+# List command
+@client.slash_command()
+@application_checks.has_role(config.dj_role_name)
+async def list(
+    interaction: Interaction,
+    list_channel: Optional[TextChannel] = SlashOption(required=False),
+) -> None:
+    """Download and list all media sent in a chosen text channel."""
+    bc = get_controller(interaction.guild_id)
```

Comment:
Note that `interaction.guild_id` will be `None` here (and on other commands) if this interaction came in via a DM.

We probably want to just disable slash commands via DM? (via the [`dm_permission`](https://docs.nextcord.dev/en/stable/api.html#nextcord.Client.slash_command) argument of `@slash_command`).

Reply:
`@application_checks.has_role(config.dj_role_name)` automatically disables this command (and all the others) for DMs since it requires a role.